### PR TITLE
Unify #inspect output formatting in several places

### PR DESCRIPTION
**Pull request by Sija**
_Reference_: `#<Type …>`
_Struct_: `Type(…)`

### CODE REVIEW

File: src/time/location.cr [hunk 1]
```diff
@@ -61,11 +61,11 @@ class Time::Location
     end
 
     def inspect(io : IO)
-      io << "Time::Zone<"
+      io << "Time::Zone("
       io << offset
       io << ", " << name
```
Comment (j8r) [inline src/time/location.cr]: Since we are here, they can be merged to `io << "Time::Zone" << offset << ", " << name`

File: src/time/location.cr [hunk 2]
```diff
@@ -74,12 +74,12 @@ class Time::Location
     getter? standard, utc
 
     def inspect(io : IO)
-      io << "Time::ZoneTransition<"
+      io << "Time::ZoneTransition("
       io << '#' << index << ", "
```
Comment (j8r) [inline src/time/location.cr]: Ditto

File: src/regex/match_data.cr [hunk 3]
```diff
@@ -332,21 +332,21 @@ class Regex
     def to_s(io : IO)
       name_table = @regex.name_table
 
-      io << "#<Regex::MatchData"
+      io << "Regex::MatchData("
       size.times do |i|
-        io << " "
+        io << " " if i > 0
```
Comment (j8r) [inline src/regex/match_data.cr]: I think the space string can be replaced by a char `' '`

File: src/file/stat.cr [hunk 4]
```diff
@@ -100,8 +100,8 @@ class File
     end
 
     def inspect(io)
-      io << "#<File::Stat"
-      io << " dev=0x"
+      io << "File::Stat("
+      io << "dev=0x"
```
Comment (j8r) [inline src/file/stat.cr]: Can be merged to `io << "File::Stat(dev=0x"`

File: src/regex/match_data.cr [hunk 5]
```diff
@@ -332,21 +332,21 @@ class Regex
...
         io << name_table.fetch(i, i) << ":" if i > 0
         self[i]?.inspect(io)
       end
-      io << ">"
+      io << ")"
```
Comment (j8r) [inline src/regex/match_data.cr]: Ditto, can be a char

File: src/regex/match_data.cr [hunk 6]
```diff
@@ -332,21 +332,21 @@ class Regex
     def to_s(io : IO)
       name_table = @regex.name_table
 
-      io << "#<Regex::MatchData"
+      io << "Regex::MatchData("
       size.times do |i|
-        io << " "
+        io << " " if i > 0
         io << name_table.fetch(i, i) << ":" if i > 0
```
Comment (j8r) [inline src/regex/match_data.cr]: `":"` => `':'`

File: src/file/stat.cr (hunk 4 above)
Comment (Sija) [reply src/file/stat.cr]: Nope, that just looks weird, especially since there's many lines below having such format.

Comment (straight-shoota) [reply src/file/stat.cr]: Why weird? Aesthetics? I don't see why these strings should not be combined.

File: src/regex/match_data.cr [hunk 7]
```diff
@@ -332,21 +332,21 @@ class Regex
     def to_s(io : IO)
       name_table = @regex.name_table
 
-      io << "#<Regex::MatchData"
+      io << "Regex::MatchData("
       size.times do |i|
-        io << ' '
+        io << ' ' if i > 0
```
Comment (straight-shoota) [inline src/regex/match_data.cr]: Could be combined with the following line to avoid `i > 0` twice.

File: src/regex/match_data.cr [hunk 8]
```diff
@@ -332,21 +332,21 @@ class Regex
...
     end
 
     def pretty_print(pp) : Nil
       name_table = @regex.name_table
 
-      pp.surround("#<Regex::MatchData", ">", left_break: nil, right_break: nil) do
+      pp.surround("Regex::MatchData(", ")", left_break: nil, right_break: nil) do
         size.times do |i|
-          pp.breakable
+          pp.breakable if i > 0
```
Comment (straight-shoota) [inline src/regex/match_data.cr]: Put this line in the `else` branch to avoid comparing twice.

File: src/time/location.cr (hunk 1 above)
Comment (straight-shoota) [reply src/time/location.cr]: @j8r I don't think that would be an improvement.

**Comment by Sija** [comment]
@asterite rebased.

**Comment by RX14** [comment]
Is it really only regex match data? cool

**Pull request merged**